src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/upsertTranslation.hpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Wazuh Vulnerability scanner - Database Feed Manager
+ * Copyright (C) 2015, Wazuh Inc.
+ * May 1, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _UPSERT_TRANSLATION_HPP
+#define _UPSERT_TRANSLATION_HPP
+
+#include "flatbuffers/flatbuffer_builder.h"
+#include "flatbuffers/flatbuffers.h"
+#include "flatbuffers/idl.h"
+#include "flatbuffers/util.h"
+#include "json.hpp"
+#include "rocksDBWrapper.hpp"
+#include "translation_generated.h"
+
+constexpr auto TRANSLATION_DATABASE_PATH {"queue/vd/translations"};
+const std::string FLATBUFFER_PATH {FLATBUFFER_SCHEMAS_DIR "translation.fbs"};
+const char* INCLUDE_DIRECTORIES[] = {FLATBUFFER_SCHEMAS_DIR, nullptr};
+
+/**
+ * @brief UpsertTranslation class.
+ *
+ */
+class UpsertTranslation
+{
+private:
+    /**
+     * @brief Inserts a translation entry in a rocksdb database.
+     *
+     * @param data JSON with the required information to create or update a translation entry.
+     * @param rocksDbWrapper rocksdb wrapper instance.
+     */
+    static void insertTranslationEntry(const nlohmann::json& data, Utils::RocksDBWrapper& rocksDbWrapper)
+    {
+        std::string translationEntryStr;
+        if (rocksDbWrapper.get(data.at("resource").get<std::string>(), translationEntryStr))
+        {
+            std::cerr << "Key already exists." << std::endl;
+        }
+        else if (data.contains("payload"))
+        {
+            std::string flatbufferSchemaStr;
+            bool valid = (flatbuffers::LoadFile(FLATBUFFER_PATH.c_str(), false, &flatbufferSchemaStr));
+            if (!valid)
+            {
+                std::cerr << "Couldn't open flatbuffer schema: " << FLATBUFFER_PATH << std::endl;
+            }
+
+            const std::string payload = data.at("payload").dump();
+            const std::string resource = data.at("resource").get<std::string>();
+            flatbuffers::Parser parser;
+
+            valid = (parser.Parse(flatbufferSchemaStr.c_str(), INCLUDE_DIRECTORIES) && parser.Parse(payload.c_str()));
+            if (!valid)
+            {
+                std::cerr << "Error: " << parser.error_ << std::endl;
```

Comment:
Shouldn't this method return here if parsing was not valid?

Reply:
Yeap, changed